### PR TITLE
docs(virtual-scroll): add embedded examples

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  width: 200px;
+  border: 1px solid black;
+}
+
+.example-item-detail {
+  height: 18px;
+}
+
+.example-alternate {
+  background: rgba(0, 0, 0, .2);
```
Comment (crisbeto) [inline src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.css]: I think we may end up with white text on a white background if somebody selected a dark theme in the docs site.

File: src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+/** @title Virtual scroll context variables */
+@Component({
+  selector: 'cdk-virtual-scroll-context-example',
+  styleUrls: ['cdk-virtual-scroll-context-example.css'],
+  templateUrl: 'cdk-virtual-scroll-context-example.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CdkVirtualScrollContextExample {
+  items = Array.from({length: 100000}).map((_, i) => `Item #${i}`);
```
Comment (crisbeto) [inline src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.ts]: nit: Can be reduced to `new Array(100000).fill(0).map((_, i) => `Item #${i}`)` which is slightly shorter.

Comment (mmalerba) [reply src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.ts]: Yeah I originally had that but switched it to this. This feels more readable to me, so I think its better for the purposes of an example.

File: src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.css (hunk 1 above)
Comment (mmalerba) [reply src/material-examples/cdk-virtual-scroll-context/cdk-virtual-scroll-context-example.css]: I don't think our examples really support theme styles, but I'll use a 50% gray so it at least looks ok in either.

**Pull request merged**